pkg/storegateway/chunkscache/cache.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package chunkscache
+
+import (
+	"context"
+
+	gotkit_log "github.com/go-kit/log"
+	"github.com/oklog/ulid"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+type Range struct {
+	BlockID   ulid.ULID
+	Start     chunks.ChunkRef
+	NumChunks int
+}
+
+type ChunksCache interface {
+	FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte)
+	StoreChunks(ctx context.Context, userID string, r Range, v []byte)
+}
+
+type TracingCache struct {
+	C ChunksCache
+}
+
+func (c TracingCache) FetchMultiChunks(ctx context.Context, userID string, ranges []Range) (hits map[Range][]byte) {
+	l := spanlogger.FromContext(ctx, gotkit_log.NewNopLogger())
+	hits = c.C.FetchMultiChunks(ctx, userID, ranges)
+
+	l.LogFields(
+		log.String("name", "ChunksCache.FetchMultiChunks"),
+		log.Int("requested", len(ranges)),
+		log.Int("hits", len(hits)),
+		log.Int("hits_bytes", hitsSize(hits)),
+		log.Int("misses", len(ranges)-len(hits)),
```

Comment:
[nit] I would clarify the subject here.

```suggestion
		log.Int("ranges_requested", len(ranges)),
		log.Int("ranges_hits", len(hits)),
		log.Int("ranges_hits_bytes", hitsSize(hits)),
		log.Int("ranges_misses", len(ranges)-len(hits)),
```